zkvm/src/ops.rs

```diff
@@ -277,3 +279,18 @@ impl Instruction {
         }
     }
 }
+
+impl Program {
+    fn new() -> Self {
+        Program(vec![])
+    }
+
+    fn push<'a, T: Into<Data>>(&'a mut self, data: T) -> &'a mut Program {
```

Comment:
Are you sure `'a` is not deduced from `&self` automatically by the compiler?

```suggestion
    fn push<T: Into<Data>>(&mut self, data: T) -> &mut Program {
```